### PR TITLE
[Chat] Rename ImageGallery to ImageOverlay and some other API changes

### PR DESCRIPTION
**Pull request by Leah-Xia-Microsoft**
# What

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/communication-react/review/beta/communication-react.api.md [hunk 1]
```diff
@@ -2998,29 +2998,22 @@ export interface _Identifiers {
...
-    titleIcon?: JSX.Element;
-}
-
-// @beta
-export interface ImageGalleryProps {
-    images: Array<ImageGalleryImageProps>;
+    imageSrc: string;
     isOpen: boolean;
     onDismiss: () => void;
+    onDownloadButtonClicked: (imageSrc: string) => void;
```
Comment (JoshuaLai) [inline packages/communication-react/review/beta/communication-react.api.md:3009]: Can introduce this in the future, but provide the ability to configure whether or not users will even see the download button

File: packages/react-components/src/components/ImageGallery.test.tsx [hunk 2]
```diff
@@ -24,92 +24,72 @@ describe('ImageGallery default layout tests', () => {
...
-    expect(image.src).toContain(imageInfo2.imageUrl);
-    expect(image.alt).toBe(imageInfo2.altText);
-
-    expect(title).toBeTruthy();
-    expect(titleIcon).toBeTruthy();
-  });
 
-  /* @conditional-compile-remove(image-gallery) */
+  /* @conditional-compile-remove(image-overlay) */
   test('It should call the onDismiss handler when the close icon is clicked', async () => {
```
Comment (JoshuaLai) [inline packages/react-components/src/components/ImageGallery.test.tsx:74]: Do we still have the functionality to close the overlay when users taps outside of the image?

File: packages/react-components/src/components/ImageGallery.tsx [hunk 3]
```diff
@@ -24,128 +24,119 @@ import {
...
-  const image = images[startIndex];
   const renderHeaderBar = (): JSX.Element => {
     return (
       <Stack className={mergeStyles(headerStyle)}>
         <Stack className={mergeStyles(titleBarContainerStyle)}>
-          {image?.titleIcon}
-          <Stack.Item className={mergeStyles(titleStyle(theme))} aria-label={image?.title}>
-            {image?.title}
+          {titleIcon}
+          <Stack.Item className={mergeStyles(titleStyle(theme))} aria-label={title || 'image'}>
```
Comment (JoshuaLai) [inline packages/react-components/src/components/ImageGallery.tsx:112]: nit: Lets fix up the casing `'Image'`

File: packages/react-components/src/components/ImageGallery.tsx [hunk 4]
```diff
@@ -24,128 +24,119 @@ import {
...
           </Stack.Item>
         </Stack>
         <Stack className={mergeStyles(controlBarContainerStyle)}>
           <DefaultButton
             className={mergeStyles(downloadButtonStyle(theme))}
-            /* @conditional-compile-remove(image-gallery) */
+            /* @conditional-compile-remove(image-overlay) */
             text={localeStrings.downloadButtonLabel}
-            onClick={() => onImageDownloadButtonClicked(image?.imageUrl || '', image?.downloadFilename || 'image')}
+            onClick={() => onDownloadButtonClicked(imageSrc || '')}
```
Comment (JoshuaLai) [inline packages/react-components/src/components/ImageGallery.tsx:121]: `imageSrc` is required right? so why do we need this `||` condition in here?

File: packages/react-components/src/localization/locales/ar-SA/strings.json [hunk 5]
```diff
@@ -31,7 +31,7 @@
   "mentionPopover": {
     "mentionPopoverHeader": "الاقتراحات"
   },
-  "imageGallery": {
+  "ImageOverlay": {
```
Comment (JoshuaLai) [inline packages/react-components/src/localization/locales/ar-SA/strings.json:34]:
```suggestion
  "imageOverlay": {

```

File: packages/react-components/src/localization/locales/de-DE/strings.json [hunk 6]
```diff
@@ -31,7 +31,7 @@
   "mentionPopover": {
     "mentionPopoverHeader": "Vorschläge"
   },
-  "imageGallery": {
+  "ImageOverlay": {
```
Comment (JoshuaLai) [inline packages/react-components/src/localization/locales/de-DE/strings.json:34]:
```suggestion
  "imageOverlay": {

```

File: packages/react-composites/src/composites/ChatComposite/ChatScreen.tsx [hunk 7]
```diff
@@ -133,12 +133,13 @@ export const ChatScreen = (props: ChatScreenProps): JSX.Element => {
...
-  /* @conditional-compile-remove(image-gallery) */
+  /* @conditional-compile-remove(image-overlay) */
   const [fullSizeAttachments, setFullSizeAttachments] = useState<Record<string, string>>({});
-  /* @conditional-compile-remove(image-gallery) */
-  const [galleryImages, setGalleryImages] = useState<Array<ImageGalleryImageProps>>([]);
-  /* @conditional-compile-remove(image-gallery) */
-  const [isImageGalleryOpen, setIsImageGalleryOpen] = useState<boolean>(false);
+  /* @conditional-compile-remove(image-overlay) */
+  const [overlayImageItem, setOverlayImageItem] =
+    useState<{ imageSrc: string; title: string; titleIcon: JSX.Element; downloadFilename: string }>();
```
Comment (JoshuaLai) [inline packages/react-composites/src/composites/ChatComposite/ChatScreen.tsx:140]: `{ imageSrc: string; title: string; titleIcon: JSX.Element; downloadFilename: string }` should we typealias this?

File: packages/react-components/src/components/ImageGallery.test.tsx (hunk 2 above)
Comment (Leah-Xia-Microsoft) [reply packages/react-components/src/components/ImageGallery.test.tsx:74]: > Do we still have the functionality to close the overlay when users taps outside of the image?

Yes this functionality remain unchanged

**Review by JoshuaLai — APPROVED** [review]

**Review by emlynmac — APPROVED** [review]

**Pull request merged**